src/s3/getobject.cpp

```diff
@@ -98,8 +154,11 @@ asio::awaitable<void> irods::s3::actions::handle_getobject(
             std::streampos current, size;
             while (d.good()) {
                 response.result(beast::http::status::ok);
-                d.read(buf_vector.data(), write_buffer_size);
+
+                std::size_t read_length = write_buffer_size < range_end+1 - offset ? write_buffer_size : range_end+1 - offset;
```

Comment:
The ternary expression is a little difficult to read.

Consider the following:
- Using parens to group expressions
- Or changing it to use an if-ladder

Reply:
I broke it into multiple lines and added comments.